External/Plugins/ASCompletion/Completion/ASGenerator.cs

```diff
@@ -2081,13 +2081,15 @@ private static void GenerateVariableJob(GeneratorJobType job, ScintillaControl s
             FlagType kind = job.Equals(GeneratorJobType.Constant) ? FlagType.Constant : FlagType.Variable;
             // evaluate, if the variable (or constant) should be generated in other class
             ASResult varResult = ASComplete.GetExpressionType(sci, sci.WordEndPosition(sci.CurrentPos, true));
+            var memberIsStatic = member != null && (member.Flags & FlagType.Static) > 0;
             if (ASContext.CommonSettings.GenerateScope && !varResult.Context.Value.Contains(ASContext.Context.Features.dot))
             {
                 position = sci.CurrentPos;
                 var start = sci.WordStartPosition(position, true);
                 var length = sci.MBSafeTextLength(contextToken);
                 sci.SetSel(start, start + length);
-                var text = "this." + contextToken;
+                var scope = memberIsStatic ? inClass.QualifiedName : "this";
+                var text = string.Format("{0}.{1}", scope, contextToken);
```

Comment:
I suggest never using `string.Format()` in completion/parser codes. They are extremely slow.
Use `scope + "." + contextToken` or `string.Concat(scope, ".", contextToken)` instead.

Also, wouldn't `ASContext.Context.Features.dot` be better than `"."`?